tests/Fixtures/Query/CustomQuery.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sonata\AdminBundle\Tests\Fixtures\Query;
+
+use Sonata\AdminBundle\Datagrid\ProxyQueryInterface;
+
+final class CustomQuery implements ProxyQueryInterface
+{
+    /**
+     * @var mixed
+     */
+    protected $sortOrder;
+
+    /**
+     * @var string
+     */
+    protected $sortBy;
+
+    public function __call($name, $args)
+    {
+        // TODO: Implement __call() method.
+    }
+
+    public function execute(array $params = [], $hydrationMode = null)
+    {
+        // TODO: Implement execute() method.
+    }
+
+    public function getSortBy()
+    {
+        return $this->sortBy;
+    }
+
+    public function setSortBy($parentAssociationMappings, $fieldMapping)
+    {
+        $alias = $this->entityJoin($parentAssociationMappings);
+        $this->sortBy = $alias.'.'.$fieldMapping['fieldName'];
+
+        return $this;
+    }
+
+    public function getSortOrder()
+    {
+        return $this->sortOrder;
+    }
+
+    public function setSortOrder($sortOrder)
+    {
+        $this->sortOrder = $sortOrder;
+
+        return $this;
+    }
+
+    public function getSingleScalarResult()
+    {
+        // TODO: Implement getSingleScalarResult() method.
```

Comment:
Please, don't return `void` in a method intended to return a value. Return a dummy value or throw a `\BadMethodCallException()` instead.

Reply:
This class is unnecessary right now. Take a look at 26517feb commit.